server.go

```diff
@@ -52,11 +52,24 @@ func parseIdentifier(path, endpoint string) (string, error) {
 	return url.PathUnescape(strings.TrimPrefix(path, endpoint+"/"))
 }
 
-// Server represents a SCIM server which implements the HTTP-based SCIM protocol that makes managing identities in multi-
-// domain scenarios easier to support via a standardized service.
+// Server represents a SCIM server which implements the HTTP-based SCIM protocol
+// that makes managing identities in multi-domain scenarios easier to support via a standardized service.
 type Server struct {
-	Config        ServiceProviderConfig
-	ResourceTypes []ResourceType
+	config        ServiceProviderConfig
+	resourceTypes []ResourceType
```

Comment:
The `config` and `resourceTypes` are mandatory, so I am not sure that it makes sense to add these as server options. It is not really an option.

Reply:
Yeah, you are probably right. I saw that there were cases in tests when empty structures were passed as configs and resources when creating the Server, so I thought they were optional, but apparently, you are right, they are required.

But if so, should the server creation fail if the required arguments were not passed? Something like this:

```golang
type ServerArgs struct {
    ServiceProviderConfig *ServiceProviderConfig
    ResourceTypes []ResourceType
}

func NewServer(args *ServerArgs, opts ...ServerOption) (*Server, error) {
        if args == nil {
                return nil, fmt.Errorf("arguments not provided")
        }

        if args.ServiceProviderConfig == nil {
                return nil, fmt.Errorf("service provider config not provider")
        }

        // the rest of the arguments checks
        // ...

	s := &Server{
		log: &noopLogger{},
	}

	for _, opt := range opts {
		opt(s)
	}

	return s, nil
}
```
By the way, how do you look at using server reference receivers instead of instance receivers?
```golang 
// changing from this
func (s Server)

// to this
func (s *Server)
```

Reply:
The receivers is just a personal preference, we've used a lot on value receivers in the package just to prevent any unwanted sharing of structs.

Since we've done this everywhere, I'd rather keep this consistent.

Reply:
Got it. What about the server constructor returning an error if mandatory arguments are missing?

Reply:
@q-uint will update the PR tomorrow with the new constructor proposal then.